### PR TITLE
[goecharger] Added tma1, tma2 and powerAll

### PR DESCRIPTION
**Pull request by chilobo**
The new version V3 of GO-E does not support the single field temperature anymore and replaced it by tma1 and tma2.
Those fields are available in version V2 too.
Added powerAll from nrg[11] which should show the combined power of all three phases.
Changed usage of amp to amx because this is more gentle to the EEPROM.

The compiled jar can be found at:
[https://www.dropbox.com/s/fzx819g2skgem0m/org.openhab.binding.goecharger-3.1.0-SNAPSHOT.jar?dl=0]

Please try if this version works with V2.

With V3 I get errors when using it because after issuing a change of current on phase 1 the code 
(senddata around line 360 in GoEChargerHandler.java )
produces an error:

`14:07:20.820 [DEBUG] [er.internal.handler.GoEChargerHandler] - POST URL = http://192.168.178.82/mqtt?payload=alw=0
14:07:20.821 [INFO ] [openhab.event.ItemStateChangedEvent  ] - Item 'GoeCharger_AllowCharging' changed from ON to OFF
14:07:21.051 [DEBUG] [er.internal.handler.GoEChargerHandler] - POST Response: Request method for this URI is not handled by server
14:07:21.055 [INFO ] [hab.event.ThingStatusInfoChangedEvent] - Thing 'goecharger:goe:GOE' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
`

This is my first submit to git, so give me feedback about my errors, omissions, handling ...

### CODE REVIEW

File: bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/api/GoEStatusResponseDTO.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
     @SerializedName("ast")
     public Integer accessConfiguration;
 
-    @SerializedName("amp")
+    @SerializedName("amx")
```
Comment (SamuelBrucksch) [inline bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/api/GoEStatusResponseDTO.java:33]: I would not replace it, as that disables the ability, to set a value, that should still be available after restart. Better would be to add it as a second property and recommend it in the PV charge sample in documentation.

**Comment by SamuelBrucksch** [comment]
Unfortunately i do not own a go-echarger anymore, so i can not test v2, maybe find someone in goingelectric forum to do so, there was a discussion about `amp` vs `amx` property already. However i would not remove it, but better add an additional property.

**Comment by chilobo** [comment]
> Unfortunately i do not own a go-echarger anymore, so i can not test v2, maybe find someone in goingelectric forum to do so, there was a discussion about `amp` vs `amx` property already. However i would not remove it, but better add an additional property.

Sure, makes sense. I will add it back. Anyway I have to adjust the value of powerall because the api states that it carries another factor than pl1..pl3

File: bundles/org.openhab.binding.goecharger/README.md [hunk 2]
```diff
@@ -88,9 +94,16 @@ when
...
 You can also define more advanced rules if you have multiple cars that charge with a different amount of phases.
 For example if your car charges on one phase only, you can set maxAmps to output of PV power, if your car charges on two phases you can set maxAmps to `pv output / 2`, and for 3 phases `pv output / 3`.
 In general the calculation would be ´maxAmps = pvOutput / phases`.
+
+Changes 06.21 (chilobo):
+Since V3 of the GO-E-Charger Temperature is not supported any more, GO-E added two temperatures tma1 and tma2.
+Added channel "Power over all" showing the combined power of all three phases
+Added channel "Maximum current temporary": This changes the current. But this value is not written to the EEPROM.
+
+Beware: Because of changes in V3 this binding is working for V1 and V2, but not for V3.
```
Comment (fwolter) [inline bundles/org.openhab.binding.goecharger/README.md:109]: The changelog will be available in the release message of OH 3.1, leading to this PR. So, it shouldn't be in the readme.

File: bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java [hunk 3]
```diff
@@ -108,6 +111,13 @@ private State getValue(String channelId, GoEStatusResponseDTO goeResponse) {
                     return UnDefType.UNDEF;
                 }
                 return new QuantityType<>(goeResponse.maxCurrent, Units.AMPERE);
+            case MAX_CURR_TEMP:
+                if (goeResponse.maxCurrTemp == null) {
+                    return UnDefType.UNDEF;
+                }
+                logger.debug("MAX_CURR_TEMP, amx ?", goeResponse.maxCurrTemp);
```
Comment (fwolter) [inline bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java:118]: This seems for debugging purposes only. Please remove the logging. See https://www.openhab.org/docs/developer/guidelines.html#f-logging

File: bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java [hunk 4]
```diff
@@ -198,11 +208,16 @@ private State getValue(String channelId, GoEStatusResponseDTO goeResponse) {
...
                 }
                 return new DecimalType(count);
-            case TEMPERATURE:
-                if (goeResponse.temperature == null) {
+            case TEMP_TMA1:
+                if (goeResponse.temptma == null) {
                     return UnDefType.UNDEF;
                 }
-                return new QuantityType<>(goeResponse.temperature, SIUnits.CELSIUS);
+                return new QuantityType<>(goeResponse.temptma[0], SIUnits.CELSIUS);
```
Comment (fwolter) [inline bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java:215]: You could add a check if this array is at least of the size 1. If the array is empty, this binding would crash.

File: bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java [hunk 5]
```diff
@@ -333,6 +361,7 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             default:
         }
         if (key != null && value != null) {
+            logger.debug("sendData: ", key, value);
```
Comment (fwolter) [inline bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java:364]: See above. Remove logging.

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java (hunk 3 above)
Comment (chilobo) [reply bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java:118]: I will delete the debugging lines.

File: bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java (hunk 4 above)
Comment (chilobo) [reply bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java:215]: I have to find out how to do that, this is my first submission to github.

**Comment by chilobo** [comment]
This is my first submission to github, I just used github desktop without really understanding what I do. I will read your comments, will try to understand them or find explanations.

File: bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java (hunk 4 above)
Comment (fwolter) [reply bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java:215]: You can check the size of the array beforehand with `if(goeResponse.temptma.length >= 1)`. Don't hesitate to ask if something is unclear. You can keep this pull request open for the time you're fixing the findings.

**Comment by Mr-iX** [comment]
@chilobo I would be also interested in having amx rather than amp as Ampere channel. Will you submit a new pull request?

**Pull request closed without merging**